### PR TITLE
Setup pipelines so they can be ran with release Configurations

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

This PR is preparation for running UI tests in Release mode and changes the *DefaultBuildConfiguration* YAML parameter to be settable at the pipeline level so that we can run builds in Release mode

This modifies the Android ControlGallery so when running as release mode it uses r8/d8 and full linking as well.
### Platforms Affected ### 
- iOS
- Android

### Testing Procedure ###
- Make sure ui tests all run
- Make sure you can still open the solution and compile it on VSMAC

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by PureWeen** [comment]
Tests keep failing to run with 
```
2019-09-15T19:30:31.0816310Z Mono Shared Runtime is not supported. This can be resolved by changing the project configuration or using a Release build.

```

Which seems like an obvious error but it's already off

https://github.com/xamarin/Xamarin.Forms/pull/7528/files#diff-f0d93698c5f1681c6d6aa0451b8f7782R56

File: build/steps/build-android.yml [hunk 1]
```diff
@@ -86,8 +86,8 @@ jobs:
       - task: CopyFiles@2
         displayName: 'Copy ${{ parameters.name }}'
         inputs:
-          SourceFolder: ${{ parameters.androidPath }}/bin/Debug/
-          Contents: '**/*.apk'
+          SourceFolder: ${{ parameters.androidPath }}/bin/Release/
+          Contents: '**/*AndroidControlGallery.apk'
```
Comment (jsuarezruiz) [inline build/steps/build-android.yml]: Is this change necessary?. With the previous code should take the .apk too, no?

**Comment by rmarinho** [comment]
Still want to merge this one @PureWeen ?

**Comment by PureWeen** [comment]
@rmarinho yes

Once we get all the upcoming things finished and merged in I'll revisit this

File: build/steps/build-android.yml [hunk 2]
```diff
@@ -81,8 +82,8 @@ jobs:
       - task: CopyFiles@2
         displayName: 'Copy ${{ parameters.name }}'
         inputs:
-          SourceFolder: ${{ parameters.androidPath }}/bin/Debug/
-          Contents: '**/*.apk'
+          SourceFolder: ${{ parameters.androidPath }}/bin/${{ parameters.buildConfiguration }}/
+          Contents: '**/*AndroidControlGallery.apk'
```
Comment (PureWeen) [inline build/steps/build-android.yml]:
```suggestion
         Contents: '**/*.apk'
```

File: build/steps/build-android.yml [hunk 3]
```diff
@@ -81,8 +82,8 @@ jobs:
       - task: CopyFiles@2
         displayName: 'Copy ${{ parameters.name }}'
         inputs:
-          SourceFolder: ${{ parameters.androidPath }}/bin/Debug/
-          Contents: '**/*.apk'
+          SourceFolder: ${{ parameters.androidPath }}/bin/${{ parameters.buildConfiguration }}/
+         Contents: '**/*.apk'
```
Comment (PureWeen) [inline build/steps/build-android.yml]:
```suggestion
          Contents: '**/*.apk'
```

File: build/steps/build-android.yml (hunk 1 above)
Comment (PureWeen) [reply build/steps/build-android.yml]: 👍
